### PR TITLE
Make apps settings tabular

### PR DESCRIPTION
**Pull request by eppfel**
Based on https://github.com/nextcloud/server/pull/2533#issuecomment-267641134 I introduced a table for the installed apps and removed Authors and description from the apps, because this view is mostly for managing your installation. (And shortened the "Enable only for specific groups" -> "Limit to groups")

I introduce a category "All installed" and set it as default. This view is sorted by
1. enabled > disabled
2. level (official > not)
3. natural (name, ...)

We have to discuss which info is essential (Link to store/docs/repo/...). For now I introduced a link to the store, but this breaks on apps, which are not from the store. How do I get this data from an App? Internal only works to get apps, which are delivered with Nextcloud.

- [ ] ~~Which info to show and how (view in store)~~
- [x] Uninstall does not remove the app properly from the DOM
- [x] icons are not perfectly in the middle, yet 
- [ ] ~~tooltip for enable-groups~~
- [x] Tests
- [ ] ~~Introduce a proper table header (we could use this to remove the "limit to groups" completely)~~

This is **only** about the tabular redesign of the installed apps. See #3194 for the complete apps management redesign.

@nextcloud/designers

### CODE REVIEW

**Comment by eppfel** [comment]
app-tabular kopie

**Comment by skjnldsv** [comment]
@eppfel Nice for an easy search. You may want to reduce the opacity of icons to 0.5 though :)

File: settings/css/settings.css [hunk 1]
```diff
@@ -508,9 +508,7 @@ span.version {
 
 #app-navigation .app-external,
 .app-version {
-	-ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=50)";
-	filter: alpha(opacity=50);
-	opacity: .5;
+	color: rgba(85,85,85,.5);
```
Comment (eppfel) [inline settings/css/settings.css]: @skjnldsv any concerns with this?

Comment (skjnldsv) [inline settings/css/settings.css]: Why not use opacity?

Comment (eppfel) [inline settings/css/settings.css]: I would like to change this to color: rgba($color, .5) with the switch to scss

Comment (eppfel) [inline settings/css/settings.css]: The border is transparent as well then

Comment (skjnldsv) [inline settings/css/settings.css]: Where will the $color var comes from?

I havent implemented the variables yet :)

Comment (eppfel) [inline settings/css/settings.css]: I do not plan to switch to scss with this PR 😁 Just was not sure, if it safe to switch

Comment (skjnldsv) [inline settings/css/settings.css]: Test and we'll see :p

Comment (skjnldsv) [inline settings/css/settings.css]: hum, since this is located outside core/css it won't work.
The function isn't implemented yet.

It's planned though :)

Comment (skjnldsv) [inline settings/css/settings.css]: Hum bis: in fact it will work. But I cant't guarantee that it will properly yet :/

**Comment by jancborchardt** [comment]
Hmm, I’m not entirely sure about this since we will want to make screenshots more important in the future. Also it corresponds to a grid view that we might want to introduce for the Files app.

What do you think here @LukasReschke @MorrisJobke?

**Comment by jancborchardt** [comment]
> I introduce a category "All installed" and set it as default.

The default view should be a new category »All apps«. Changing category just for installing apps is what’s already cumbersome right now. ;)

**Comment by eppfel** [comment]
Just to make this clear again. This is only for __installed apps__. Appstore categories will be in another PR (see #3194) and I am positive on the larger screenshots.

> Hmm, I’m not entirely sure about this since we will want to make screenshots more important in the future.

I think we have to acknowledge that there are two major use-cases. 
1. Installing new apps
    1. explorative: What (new) apps are there?
    1. targeted: I want to install x?
1. Manage apps
    1. targeted: App x is not useful/broken/etc. and I want to remove it?
    1. targeted: The updater disabled some Apps, let's see and reenable.
    1. targeted: There are updates, do them.

I think the explorative use is very seldom and you might do this in the appstore right now.
The targeted new install will probably be by a search in an "All available apps" category (see #3194 ) and 
All managing use-cases are (mostly) directed and in my experience happen more often. The tabular view is just too show more apps in one view. And screenshots not add anything here, because you already know the apps and probably remember their icons.

Mac app store for example:

appstore-explore
appstore-bought

> The default view should be a new category »All apps«. Changing category just for installing apps is what’s already cumbersome right now. ;)

"All installed": The idea is to manage installed apps in one place, rather than two (disabled / enabled).
We can introduce the "All available" in a next step and set it default then, but still I would like introduce "All installed" because with more and more apps in the store, these might disturb the user. What's your take on that @jancborchardt ?

Further. installing new apps happens less often, so another click is unproblematic IMO. And once selected the category changes,. So even if you install a bunch of new apps with a new installation, you hit "All available" once and then search the apps you need one after another.


And I need feedback on leaving out meta, like author aso.

**Comment by Espina2** [comment]
This is very good. The offical tag is just weird, it looks like a button. Can we transform this in a normal text style? @eppfel @jancborchardt

**Comment by jancborchardt** [comment]
@eppfel ok, agree on the layout. I thought about it more too, and thanks a lot for the added design reasoning. :)

> "All installed": The idea is to manage installed apps in one place, rather than two (disabled / enabled).

Installed == enabled though. We should not introduce another word or concept here. Maybe we should switch the wording from »Enabled / Not enabled« to »Installed / Not installed« (and »Install / Uninstall« on the buttons) but definitely not use both words at the same time.

> Further. installing new apps happens less often, so another click is unproblematic IMO.

Installing new apps is definitely as big of a use case as managing the current ones is. See also the existing issue: »All apps« category in apps management #126


Another thing btw talking about simplifying and streamlining the management: We should link from the list of installed apps to the settings section of each app. :) Also cc @nickvergessen

**Comment by eppfel** [comment]
That's how it looks right now.
bildschirmfoto 2017-01-23 um 17 49 41

>  The offical tag is just weird, it looks like a button. Can we transform this in a normal text style?

@Espina2 Yes, this needs rework. We might want to introduce batches/chips?
But the question is, do we need this info there anyway?

bildschirmfoto 2017-01-23 um 17 58 17

> Installed == enabled though. We should not introduce another word or concept here. Maybe we should switch the wording from »Enabled / Not enabled« to »Installed / Not install

@jancborchardt 
But technically there is a difference.
- Installed / not installed: is the app present on your installation or not
- enabled / not enabled: is the app active and can be used

I believe admins care for the difference.

After reading #126 "All installed" does exactly this (even the sorting). But I guess you want the appstore categories included as well, right?

I move the discussion about the apps management structure / navigation / categories to the overview issue #3194

**Comment by Espina2** [comment]
Another problem is the iconography used to represent the apps is not unique enough. We already use the same icons to others things... But in the other hand I know its hard to have more unique icons on this... 

Anyway its looks a lot better @eppfel .

**Comment by eppfel** [comment]
Ok, I want to get this in, so let's _warp_ things up. 

warp speed

I'll postpone:

- Which info to show and how (view in store) -> Bigger issue and should be easy to integate later into the tabular layout
- Introduce table header -> for now columns are self-explanatory
- tooltip for enable-groups -> kind of dependents on the two previous points

The two questions remaining is

#### 1. the navigation:

I propose: Drop enabled and disabled and make it into one category: "Enabled/Disabled" or "Installed". Either way, it is an improvement to have all _installed_ apps in one view. See #3194 for broader discussion.

#### 2. dynamic sorting

When enabling an app should the list be sorted and the app jump up. I say we leave the app, where it is, to not confuse the user.

**Comment by claell** [comment]
1. I agree that your proposal of the category "installed" offers the same functionality as "enabled" and "disabled". So they can be dropped.
However I'd change the name to not confuse the user to something like "downloaded". Installed means the same as enabled to me.
This clarifies the whole situation a bit more. When I first used the app store I did not know what exactly "enabled" and "disabled" is and that there can be downloaded but not enabled apps.

2. How are the apps sorted right now? Alphabetically? It might be helpful to have some order and having all enabled apps on top might lead to better usability. If this is not done dynamically I'd propose to leave the position where it is when inside the category and show it resorted on next visit.

**Comment by eppfel** [comment]
1. Maybe we have to rethink, the enabled / disabled,  installed / not installed paradigm then. E.g. Do not differentiate at all

2. As stated above, sorting is enabled -> official -> name. Opening a category triggers sorting automatically.

**Comment by claell** [comment]
2. Sorry didn't read that carefully enough. To me it makes sense to leave it that way to avoid confusing the user as you said. It probably wouldn't be good to have an app move to the top right after the user clicks on "enable".

**Comment by jancborchardt** [comment]
@eppfel any update here, or a minimum version we can merge? :) Would be cool to get this in for Nextcloud 12.

**Comment by eppfel** [comment]
> 1. the navigation:
>
>I propose: Drop enabled and disabled and make it into one category: "Enabled/Disabled" or "Installed". Either way, it is an improvement to have all installed apps in one view. See #3194 for broader discussion.

@jancborchardt This decision blocks this PR right now.

**Comment by jancborchardt** [comment]
@eppfel sorry for  the late reply!

> I propose: Drop enabled and disabled and make it into one category: "Enabled/Disabled" or "Installed"

I fully agree with that. Let’s just call it »Your apps« as other appstores do it similarly. Sort enabled apps first of course.

@eppfel is it possible to get this in for Nextcloud 12? Would be awesome to, and please ask if you need help. :)

**Comment by jancborchardt** [comment]
@eppfel I’d say let’s merge this and do further adjustments in separate pull requests … it’s getting too big and we just need to move forward. :)

**Comment by MorrisJobke** [comment]
Only two items:

* [ ] show spinner for changing the different sections in the middle (like on the initially loading)
bildschirmfoto 2017-03-28 um 18 26 12
* [ ] show a spinner for the images of the app store apps (and a placeholder for those that don't have one - otherwise it looks broken)
bildschirmfoto 2017-03-28 um 18 25 49

**Comment by jancborchardt** [comment]
Will look into the spinner.

How can I check for if an app doesn’t have an image? cc @BernhardPosselt @eppfel?

**Comment by BernhardPosselt** [comment]
The screenshots attribute on the returned JSON will be empty, see http://nextcloudappstore.readthedocs.io/en/latest/restapi.html#get-all-apps-and-releases

Unsure however how this is implemented in the server @LukasReschke

**Comment by MorrisJobke** [comment]
@jancborchardt Ping

**Comment by eppfel** [comment]
> @jancborchardt Ping

Pong.

I can look at it thursday

**Comment by eppfel** [comment]
@jancborchardt @nextcloud/designers 

Removing `limit to groups` on mobile view looks cleaner. But removing/hiding functionality completely from the user feels wrong.
bildschirmfoto 2017-04-05 um 21 00 21

I know it looks shit and I don't have another idea, right now, still...
Anyone ideas, opinions?
bildschirmfoto 2017-04-05 um 21 03 50

**Comment by jancborchardt** [comment]
@eppfel I made it so it’s visible still when there’s a bit of space and only hides when very narrow. Just because next to the disable/enable button it’s the only really important piece (cause it’s interactive). Does that not work?

**Comment by MorrisJobke** [comment]
@jancborchardt What about https://github.com/nextcloud/server/pull/3195#issuecomment-289945782 ?

**Comment by jancborchardt** [comment]
All issues fixed and rebased too. Ready for review @nextcloud/javascript @nextcloud/designers! :)

**Review by MorrisJobke — APPROVED** [review]
Tested and works :+1: I restarted the CI because JS unit tests failed with a connection timeout.

**Comment by MorrisJobke** [comment]
Mmmmh ... `"/drone/src/github.com/nextcloud/server/core/vendor/DOMPurify/dist/purify.min.js" were excluded or matched by prior matchers.`

@LukasReschke You touched this lately. Any idea?

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
Looks great!

Only two remarks:

1. "Limit to groups" is not aligned with the checkbox
bildschirmfoto von 2017-04-25 08-13-25

2. During enabling/disabling an app, the button text is wider and therefore causes the columns to change their size. This look a bit odd.

**Comment by jancborchardt** [comment]
@ChristophWurst 

> "Limit to groups" is not aligned with the checkbox

This seems to be an issue in master actually. Will see to fix via a different PR.

> During enabling/disabling an app, the button text is wider and therefore causes the columns to change their size. This look a bit odd.

Yes, but this only occurs on narrow sizes and will be variable with languages anyway. There’s no immediate way I can think of adjusting that. We should check it out after _feature_ freeze, ok? :)

**Review by ChristophWurst — APPROVED** [review]
Fine by me

**Comment by LukasReschke** [comment]
> Mmmmh ... "/drone/src/github.com/nextcloud/server/core/vendor/DOMPurify/dist/purify.min.js" were excluded or matched by prior matchers.
>
> @LukasReschke You touched this lately. Any idea?

Not a lot of ideas since it works locally but https://github.com/nextcloud/server/pull/4486 could be it.

**Comment by MorrisJobke** [comment]
Okay - the JSUnit failure is unrelated -> merge it.

**Comment by MariusBluem** [comment]
Awesome 🎉🙈

**Comment by jancborchardt** [comment]
Awesome work @eppfel! :tada:

**Pull request merged**